command-service/src/main.rs

```
@@ -1,41 +1,85 @@
+async fn main() {
     env_logger::init();
-    let args = Args::from_args();
+    let args: Args = Args::from_args();
 
     trace!("Environment: {:?}", args);
 
-    let report_service = ReportService::new(args.report_config).await?;
-
-    let output_service = Output::new(args.output_config).await?;
-
-    let message_router = MessageRouter::new(
-        output_service.channel(),
-        report_service,
-        output_service.name(),
-    );
+    metrics::serve();
+    match args.output_config {
+        OutputArgs::Sleigh(sleigh_config) => {
+            start_services(
+                args.input_config,
+                args.report_config,
+                SleighOutputPlugin::new(sleigh_config)
+                    .await
+                    .expect("Initializing Sleigh output interface"),
+            )
+            .await
+        }
+        OutputArgs::Postgres(postgres_config) => {
+            start_services(
+                args.input_config,
+                args.report_config,
+                PostgresOutputPlugin::new(postgres_config)
+                    .await
+                    .expect("Initializing Postgres output interface"),
+            )
+            .await
+        }
+        OutputArgs::Druid(druid_config) => {
+            start_services(
+                args.input_config,
+                args.report_config,
+                DruidOutputPlugin::new(druid_config)
+                    .await
+                    .expect("Initializing Druid output interface"),
+            )
+            .await
+        }
+        OutputArgs::VictoriaMetrics(victoria_metrics_config) => {
+            start_services(
+                args.input_config,
+                args.report_config,
+                VictoriaMetricsOutputPlugin::new(victoria_metrics_config)
+                    .expect("Initializing Victoria Metrics output interface"),
+            )
+            .await
+        }
+    }
+    .expect("Running services")
```

Why not just return an error from `main`?

---

I guess it wasn't there when I started this feature. We need to standarize this, we have some apps unwrap and some `?`

---

Changed to anyhow